client/cmd/dexc-desktop/main.go

```diff
@@ -0,0 +1,556 @@
+	const HWND hDesktop = GetDesktopWindow();
+	GetWindowRect(hDesktop, &desktop);
+	return desktop.bottom;
+}
+#else
+#include <X11/Xlib.h>
+int display_width() {
+	Display* d = XOpenDisplay(NULL);
+	Screen*  s = DefaultScreenOfDisplay(d);
+	return s->width;
+}
+int display_height() {
+	Display* d = XOpenDisplay(NULL);
+	Screen*  s = DefaultScreenOfDisplay(d);
+	return s->height;
+}
+#endif
+*/
+import "C"
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"runtime"
+	"runtime/debug"
+	"runtime/pprof"
+	"sync"
+	"syscall"
+	"time"
+
+	"decred.org/dcrdex/client/app"
+	_ "decred.org/dcrdex/client/asset/bch" // register bch asset
+	_ "decred.org/dcrdex/client/asset/btc" // register btc asset
+	_ "decred.org/dcrdex/client/asset/dcr" // register dcr asset
+	_ "decred.org/dcrdex/client/asset/ltc" // register ltc asset
+
+	// Ethereum loaded in client/app/importlgpl.go
+
+	"decred.org/dcrdex/client/core"
+	"decred.org/dcrdex/client/rpcserver"
+	"decred.org/dcrdex/client/webserver"
+	"decred.org/dcrdex/dex"
+	"github.com/webview/webview"
+
+	"fyne.io/systray"
+	"github.com/gen2brain/beeep"
+)
+
+const appName = "dexc"
+
+var (
+	log     dex.Logger
+	exePath = findExePath()
+	srcDir  = filepath.Join(filepath.Dir(exePath), "src")
+)
+
+func main() {
+	// Wrap the actual main so defers run in it.
+	err := mainCore()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func mainCore() error {
+	appCtx, cancel := context.WithCancel(context.Background())
+	defer cancel() // don't leak on the earliest returns
+
+	// Parse configuration.
+	cfg, err := configure()
+	if err != nil {
+		return fmt.Errorf("configration error: %w", err)
+	}
+
+	// A single process cannot run multiple webview windows, so we run webview
+	// as a subprocess. We could create a simpler webview binary to call that
+	// would be substantially smaller than the dexc binary, but when done that
+	// way, the opened window does not have the icon that the system associates
+	// with dexc and taskbar icons won't be stacked. Instead, we'll create a
+	// short path here and execute ourself with the --webview flag.
+	if cfg.Webview != "" {
+		runWebview(cfg.Webview)
+		return nil
+	}
+
+	// Initialize logging.
+	utc := !cfg.LocalLogs
+	if cfg.Net == dex.Simnet {
+		utc = false
+	}
+	logMaker, closeLogger := app.InitLogging(cfg.LogPath, cfg.DebugLevel, cfg.LogStdout, utc)
+	defer closeLogger()
+	log = logMaker.Logger("APP")
+	log.Infof("%s version %s (Go version %s)", appName, app.Version, runtime.Version())
+	if utc {
+		log.Infof("Logging with UTC time stamps. Current local time is %v",
+			time.Now().Local().Format("15:04:05 MST"))
+	}
+
+	// The --kill flag is a backup measure to end a background process (that
+	// presumably has active orders).
+	if cfg.Kill {
+		sendKillSignal(cfg.AppData)
+		return nil
+	}
+
+	startServer, quit, err := synchronize(cfg.AppData)
+	if err != nil || quit {
+		return err
```

Comment:
If err is nil here, we expect the sync server to signal to the running instance to revive it's window.
Hopefully there still is a running dexc-instance.